packages/cli/src/index.ts

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env node
+import chalk from "chalk";
+import { Command } from "commander";
+import * as figlet from "figlet";
+import { getToken, writeTokenFile } from "./auth";
+import { check } from "./check";
+import { ABBY_BASE_URL, getTokenFilePath } from "./consts";
+import { pullAndMerge } from "./pull";
+import { push } from "./push";
+import { ConfigOption, HostOption } from "./sharedOptions";
+
+const program = new Command();
+
+console.log(chalk.magenta(figlet.textSync("abby-cli", { horizontalLayout: "full" })));
+
+program.name("abby-cli").description("CLI Tool for Abby").version("0.0.1");
+
+program
+  .command("login")
+  .option("-t, --token <token>", "token")
+  .action(async ({ token }) => {
+    if (typeof token === "string") {
+      await writeTokenFile(token);
+      console.log(chalk.green(`Token successfully written to ${getTokenFilePath()}`));
+    } else {
+      console.log(
+        chalk.red(`You need to provide a token to log in.`),
+        chalk.green(`\nYou can get one at ${ABBY_BASE_URL}profile`)
+      );
+    }
+  });
+
+program
+  .command("pull")
+  .addOption(HostOption)
+  .addOption(ConfigOption)
+  .action(async (options: { config?: string; host?: string }) => {
+    try {
+      const token = await getToken();
+      await pullAndMerge({
+        apiKey: token,
+        apiUrl: options.host,
+        configPath: options.config,
+      });
+    } catch (e) {
+      console.error(e);
+    }
+  });
+
+program
+  .command("push")
+  .description("push local config to server")
+  .addOption(HostOption)
+  .addOption(ConfigOption)
+  .action(async (options: { config?: string; host?: string }) => {
+    try {
+      const token = await getToken();
+      await push({ apiKey: token, apiUrl: options.host, configPath: options.config });
+    } catch (e) {
+      console.log(chalk.red("Please login first"));
```

Comment:
we tell the user in "any error case" to log in first? maybe this wasnt the reason? =)